source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -446,6 +453,23 @@ void Filter::closeStream() {
   }
 }
 
+void Filter::halfCloseAndWaitForRemoteClose() {
+  if (!config_->grpcService().has_value()) {
+    return;
+  }
+
+  if (stream_) {
+    ENVOY_STREAM_LOG(debug, "Calling half-close on stream", *decoder_callbacks_);
+    if (stream_->halfCloseAndDeleteOnRemoteClose()) {
```

Comment:
I am wondering why not add if (config_->gracefulGrpcClose()) check inside closeStream() at above line 446?  That way, we do not need to add such logic in every caller place of closeStream()?

Reply:
Some places in ext_proc need to call close() that will reset the stream, i.e. when response was wrong or timeout occurred - in this case resetting the stream is correct.

There are so far only two places that need to close gracefully - when server sent immediate reply command or at the very end - when protocol completed gracefully. Only these two places use the new graceful close.

Reply:
SGTM